examples/ngsi_v2/e11_ngsi_v2_context_specific_models.py

```diff
@@ -0,0 +1,151 @@
+# creation.
+class AddressAttribute(ContextAttribute):
+    """
+    Context attribute for address
+    """
+
+    def __init__(self, type: str = None, **data):
+        if type is None and self.model_fields["type"].default:
+            type = self.model_fields["type"].default
+        super().__init__(type=type, **data)
+
+    # add default for type if not explicitly set
+    type: Literal["PostalAddress"] = FieldInfo.merge_field_infos(
+        # First position is the field info of the parent class
+        ContextAttribute.model_fields["type"],
+        # set the default value
+        default="PostalAddress",
+        # overwrite the description
+        description="Type of the Address",
+        # freeze the field if necessary, e.g. if you want to prevent the user
+        # from changing the type after creation.
+        frozen=True,
+        # for more options see the pydantic documentation
+    )
+    # add the value model. If you want to enforce the value to be set you can
+    # remove the default value
+    value: AddressModel = None
+
+
+# Define a context specific model for a weather station
+# First, inherit from the ContextEntity class, where each additional attribute
+# must be of type ContextAttribute
+class WeatherStation(ContextEntity):
+    """
+    A context specific model for a weather station
+    """
+    # add default for type if not explicitly set
+    type: str = FieldInfo.merge_field_infos(
+        # First position is the field info of the parent class
+        ContextEntity.model_fields["type"],
+        # set the default value
+        default="CustomModels:WeatherStation",
+        # overwrite the title in the json-schema if necessary
+        title="Type of the Weather Station",
+        # overwrite the description
+        description="Type of the Weather Station",
+        # validate the default value if necessary
+        validate_default=True,
+        # freeze the field if necessary
+        frozen=True,
+        # for more options see the pydantic documentation
+    )
+    # add model specific attributes including their type.
+    temperature: ContextAttribute = ContextAttribute(type="Number")
+    humidity: ContextAttribute = ContextAttribute(type="Number")
+    pressure: ContextAttribute = ContextAttribute(type="Number")
+    # add the customized address attribute with an empty value
+    # how strict validation will be performed depends on the used pydantic
+    # address model. Hence, the user can define the validation rules for it and
+    # ship them with the schema file.
+    address: AddressAttribute = AddressAttribute()
+
+
+if __name__ == "__main__":
+    # Now we can create the weather station model and export it to a
+    # json-schema file without explicitly defining the entity-type.
+    # Furthermore, we can use the model to create a new weather station entity
+    weather_station = WeatherStation(
+        id="myWeatherStation",
+        type="brick:WeatherStation",
```

Comment:
I would consider just using "WeatherStation" here, because "brick" is an undefined namespace, and here it is not JSON-LD

Reply:
That is fine with me